core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -313,6 +316,23 @@ protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startin
   protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingSnapshotId,
                                                   Expression dataFilter, Iterable<DataFile> dataFiles,
                                                   boolean caseSensitive) {
+    validateNoNewDeletesForDataFiles(base, startingSnapshotId, dataFilter, dataFiles, caseSensitive, false);
+  }
+
+  /**
+   * Validates that no new delete files that must be applied to the given data files have been added to the table since
+   * a starting snapshot, with the option to ignore equality deletes during the validation.
```

Comment:
Nbd, but I would add a note here about why we want to ignore equality deletes, just so future readers could understand.